lib/Graphs/SVFG.cpp

```diff
@@ -524,6 +544,25 @@ void SVFG::dump(const std::string& file, bool simple)
     GraphPrinter::WriteGraphToFile(outs(), file, this, simple);
 }
 
+std::set<const SVFGNode*> SVFG::fromLLVMValue(const llvm::Value* value) const
+{
+    PAG* pag = PAG::getPAG();
+    std::set<const SVFGNode*> ret;
+    // search for all PAGEdges first
+    for (const PAGEdge* pagEdge : pag->getValueEdge(value)) {
+        PAGEdgeToStmtVFGNodeMapTy::const_iterator it = PAGEdgeToStmtVFGNodeMap.find(pagEdge);
+        if (it != PAGEdgeToStmtVFGNodeMap.end()) {
+            ret.emplace(it->second);
+        }
+    }
+    // add all PAGNodes
+    PAGNode* pagNode = pag->getPAGNode(pag->getValueNode(value));
+    if(hasDef(pagNode)) {
+        ret.emplace(getDefSVFGNode(pagNode));
+    }
+    return ret;
```

Comment:
We do not return the local set. Pass a reference set as a parameter?

Reply:
The set is moved in these cases, so there shouldn't be a problem.

Reply:
If you make this function public, it may be accidentally used by other users. I would suggest making the function like:
 `void SVFG::fromLLVMValue(const llvm::Value* value, std::set<const SVFGNode*>& nodeset) const;'  where `nodeset` is a reference referring to set in the caller, so that we will not return a local set.

Reply:
Here applies what I said in the other comment. Returning a set or specifying a reference parameter leads to the same code.

Theoretical, the local set is returned by a copy. However, in practice it is moved, see https://isocpp.org/blog/2013/02/no-really-moving-a-return-value-is-easy-stackoverflow